### ISSUE TITLE
Document or automate modifying package.json when deploying a sub set of supported brands

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/chipper#1062]
It was recently discussed in a dev meeting that when creating a new release branch for a sim that supports multiple brands, but only publishing a sub set of those brands, that you need to modify package.json to remove the brands that are not included in the release.

For example, if supportedBrands is 
```
    "supportedBrands": [
      "phet",
      "phet-io",
      "adapted-from-phet"
    ],
```

And you run `grunt rc --brands=phet`, you still need to manually remove the "phet-io" (and potentially the "adapted-from-phet"??) entry.

I couldn't find the issue where we determined this was necessary, and didn't see it in https://github.com/phetsims/phet-info/blob/master/deployment-info/chipper-2.0.md which is where I usually go for publication steps.

Can we add this to the documentation? Or even better, could this be handled automatically by the grunt task?

**Comment by jessegreenberg** [comment]
Over slack there were some comments:

```
Jonathan Olson  1:49 PM
Just do the RC/production deploys with --brands=phet, it won't be maintenanced with phet-io

Michael Kauzmann:spiral_calendar_pad:  1:49 PM
I’m not sure of all the constraints in trying to remove a brand in a subsequent verison.
I thought it is helpful to do this in case there is an MR during RC testing, as that is the only way that the process knows what brands. Once published metadata is used to determine brands.

...

Jonathan Olson  1:50 PM
I'm fine if it's removed however
New

Michael Kauzmann:spiral_calendar_pad:  1:50 PM
I'm fine if it's removed however
Even if not totally necessary, I think it is prudent, clear, and a good thing to do.
1:51
My vote, update the documentation to make the branches supported brands match what is getting published.
SR: :+1:
```

**Comment by jessegreenberg** [comment]
More comments from slack:
```

Jonathan Olson  2:22 PM
Should we consider adjusting the brands automatically in package.json based on the grunt create-branch or initial grunt rc that creates the branch?

Michael Kauzmann  3:43 PM
It sounds right to me, like to require a brands arg to to the rc task? Want to take it up in that issue?

Sam Reid  3:44 PM
maybe better to have an assertion-style check, since if you accidentally put grunt --brands=phet maybe it should tell you that’s a discrepancy rather than change the code?
3:45
Or we could have a process that --brands is no longer supported, it always comes from the package.json.  And a prompt tells you what the package.json brands are, and prompts you to say [Y/n]

Jesse Greenberg  3:51 PM
grunt --brands=phet while there are other brands in packe.json isn't always an accident though. In my recent case I want to publish phet brand without phet-io brand, even though both were in package.json.

Michael Kauzmann  3:51 PM
but maybe only on the branch.
```

**Comment by samreid** [comment]
Summarizing the last few comments from the proposal in slack: when on a release branch, brands would be taken from package.json and not supported as a command line --brands argument (would that error out if provided or just if different than package.json brands?).  Running a build on a release branch, it would print the brands mentioned in package.json and prompt [Y/n] to continue.

**Comment by marlitas** [comment]
Pinging @jessegreenberg for status, and removing from subgroups. Please move back if desired!

**Comment by zepumph** [comment]
After the above changes, I was able to confirm the fix with the following steps:

1. Create a new release branch and RC with this command, saying I only support phet here:  `grunt rc --repo=bumper --brands=phet --branch=19.9`
2. Once completed trying to create a second RC trying to build phet-io brand `grunt rc --repo=bumper --brands=phet-io --branch=19.9`. This command failed!

I also updated the chipper 2.0 doc accordingly

Questions for review:

* What is the before default for create-release? https://github.com/phetsims/perennial/blob/786524183140c7da958bdc6ec0c1a66293186ebd/js/grunt/Gruntfile.js#L287 Should I just require it? I was scared to do that for any potential uses of create-release that aren't using `brands` right now. I think I like this default as is.
* Is it ok that we will pretty much always be stripping adapted-from-phet brand out of release branches. I am alright with it.
